### PR TITLE
Update `<title>` in  `<meta>` section in the index.html page, for all pages

### LINKED ISSUE #361: Enhancement: Use `leptos_meta` to update <title> for each page

### PR DESCRIPTION
**Pull request by n1tranquilla**
Closes #361

### CODE REVIEW

File: cypress/e2e/seo-optimization/titles.cy.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+suite(["@CI"],'meta title',() => {
+
```
Comment (robinbb) [inline cypress/e2e/seo-optimization/titles.cy.js:2]: Very nice suite!

File: src/addresses/page.rs [hunk 2]
```diff
@@ -84,9 +87,20 @@ pub fn AccountSpotlightPage() -> impl IntoView {
         },
     );
 
-    let public_key = move || memo_params_map.with(|p| p.get("id").cloned());
+    let public_key = move || memo_params_map.get().get("id").cloned();
```
Comment (robinbb) [inline src/addresses/page.rs:90]: This seems unrelated to the problem?

File: src/stakes/page.rs [hunk 3]
```diff
@@ -37,35 +34,38 @@ pub fn StakesPage() -> impl IntoView {
...
-                                    let header = if i_next_epoch == i_qs_epoch {
-                                        "Current Staking Ledger".to_string()
-                                    } else {
-                                        format!("Epoch {} Staking Ledger", i_qs_epoch)
-                                    };
-                                    (i_prev_epoch, i_next_epoch, i_curr_epoch, header)
-                                }
-                                _ => (0, 0, 0, "".to_string()),
-                            }
+                            let header = if curr_epoch == qs_epoch {
```
Comment (robinbb) [inline src/stakes/page.rs:57]: This is unrelated, no?

File: src/transactions/page.rs [hunk 4]
```diff
@@ -62,7 +65,19 @@ pub fn TransactionSpotlightPage() -> impl IntoView {
         },
     );
 
+    create_effect(move |_| {
```
Comment (robinbb) [inline src/transactions/page.rs:68]: Seems like this is unrelated to the problem to be solved, unless I misunderstand?

**Review by robinbb — CHANGES_REQUESTED** [review]

File: src/addresses/page.rs (hunk 2 above)
Comment (n1tranquilla) [reply src/addresses/page.rs:90]: It is unrelated, yes

File: src/stakes/page.rs (hunk 3 above)
Comment (n1tranquilla) [reply src/stakes/page.rs:57]: This one is related. The test revealed this logic wasn't working quite right.

File: src/transactions/page.rs (hunk 4 above)
Comment (n1tranquilla) [reply src/transactions/page.rs:68]: This sets the memo for the <title> (see test to see how memo ends up in title)

**Review by robinbb — APPROVED** [review]
